libmate-desktop/mate-desktop-item.c

```diff
@@ -1937,7 +1942,7 @@ ditem_execute (const MateDesktopItem *item,
 		g_slist_free_full (vector_list, g_free);
 
 #ifdef HAVE_STARTUP_NOTIFICATION
-		if (sn_context != NULL &&
+		if (sn_context != NULL && (GDK_IS_X11_DISPLAY (display)) &&
```

Comment:
@lukefromdc 
Sorry, i notice compile warnings during testing new PRs.
```
mate-desktop-item.c:1945:18: warning: The left operand of '!=' is a garbage value [core.UndefinedBinaryOperatorResult]

                if (sn_context != NULL && (GDK_IS_X11_DISPLAY (display)) &&

                    ~~~~~~~~~~ ^
```
https://app.travis-ci.com/github/mate-desktop/mate-desktop/jobs/607773800#L1924
When building a local rpm the warning looks like this.
```
mate-desktop-item.c:1945:20: warning: 'sn_context' may be used uninitialized [-Wmaybe-uninitialized]
 1945 |                 if (sn_context != NULL && (GDK_IS_X11_DISPLAY (display)) &&
      |                    ^
mate-desktop-item.c: In function 'mate_desktop_item_launch_on_screen_with_env':
mate-desktop-item.c:1778:28: note: 'sn_context' was declared here
 1778 |         SnLauncherContext *sn_context;
      |                            ^~~~~~~~~~
```

Reply:
This warning is gone with https://github.com/mate-desktop/mate-desktop/pull/587